### PR TITLE
Let subplot -Wpen draw optional dividing lines

### LINKED ISSUE #1403: Let subplot draw lines between the tiles
**Description of the desired feature**

The subplot module should have a **-W**_pen_ option that will draw lines centered between all neighboring tiles.  This is particularly useful with the subplot does not use **-B** to draw annotations.

### PR DESCRIPTION
**Pull request by PaulWessel**
There are times, especially with **-B+n**, where drawing lines between subplots is required.  Closes #1403.

### CODE REVIEW

**Comment by seisman** [comment]
I'm OK with the -W option for interior lines, but how to draw the exterior frame for the whole figure?

I modified your test script:
```
#!/usr/bin/env bash
# Test lines between subplot with -W
gmt begin tictactoe png
    gmt set FONT_ANNOT_PRIMARY 12p
    gmt subplot begin 3x3 -Fs5c -M1c -W1p,- -Blrbt
        gmt coast -Rg -JG30/30/?  -Gred -Bg -c0,0
        gmt coast -Rg -JG120/30/? -Gred -Bg -c1,1
        gmt coast -Rg -JG210/30/? -Gred -Bg -c2,2
    gmt subplot end
    gmt basemap -Blrbt
gmt end show
```
and get this:

image

**Comment by PaulWessel** [comment]
The **+p**_pen_ modifier to **-F** can draw the exterior lines, but I think you are looking for a way to pad that on the outside?  This opens up the question if **-M** should also affect spacing outside the subplot interiors.

**Comment by seisman** [comment]
> The **+p**_pen_ modifier to **-F** can draw the exterior lines

Yes, it works. I just forgot this modifier.

> I think you are looking for a way to pad that on the outside? This opens up the question if **-M** should also affect spacing outside the subplot interiors.

Yes, and it seems the bottom side already has a padding while other three sides don't.

**Comment by PaulWessel** [comment]
The bottom side was the bug fixed in #1406.

**Comment by PaulWessel** [comment]
I think you can get the effect you want with -C on all sides.  BTW, I think we need to allow **-C**_gap_ to mean **-Cw**_gap_ **-Ce**_gap_ **-Cs**_gap_ **-Cn**_gap_ as that is just too tedious.

**Comment by seisman** [comment]
> I think you can get the effect you want with -C on all sides. 

No, it doesn't work as expected. This is what I got with **-Cw1c -Ce1c -Cs1c -Cn1c**.

image

**Comment by seisman** [comment]
BTW, I think it make more sense to make it as a modifier of **-F** option, just like **+p** for exterior lines.

File: doc/rst/source/subplot.rst_ [hunk 1]
```diff
@@ -25,6 +25,7 @@ Synopsis (begin mode)
 [ **-S**\ *layout* ]
 [ **-T**\ *title* ]
 [ |SYN_OPT-V| ]
+[ **-W**\ *pen* ]
```
Comment (seisman) [inline doc/rst/source/subplot.rst_]:
```suggestion
```

File: doc/rst/source/subplot.rst_ [hunk 2]
```diff
@@ -105,10 +106,9 @@ Optional Arguments
 
 **-C**\ [*side*]\ /*clearance*\ [**u**\ ]
     Reserve a space of dimension *clearance* between the margin and the subplot on the specified
-    side, using *side* values from **w**, **e**, **s**, **n**, or **x** for both **w** and **e**
-    or **y** for both **s** and **n**.  The option is repeatable to set aside space
-    on more than one side.  If no *side* is given the it means the same constant clearance on all sides.
-    Such space will be left untouched by the main map plotting but can
+    side, using *side* values from **w**, **e**, **s**, or **n**, or **x** for both **w** and **e**
+    or **y** for both **s** and *nn**.  No *side* means all sides. The option is repeatable to set aside space
```
Comment (seisman) [inline doc/rst/source/subplot.rst_]:
```suggestion
    or **y** for both **s** and *n**.  No *side* means all sides. The option is repeatable to set aside space
```

File: doc/rst/source/subplot.rst_ [hunk 3]
```diff
@@ -160,6 +160,11 @@ Optional Arguments
 .. |Add_-V| unicode:: 0x20 .. just an invisible code
 .. include:: explain_-V.rst_
 
+.. _subplot_begin-W:
```
Comment (seisman) [inline doc/rst/source/subplot.rst_]: Need to be removed.

File: src/subplot.c [hunk 4]
```diff
@@ -155,8 +157,8 @@ GMT_LOCAL void Free_Ctrl (struct GMT_CTRL *GMT, struct SUBPLOT_CTRL *C) {	/* Dea
 GMT_LOCAL int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s begin <nrows>x<ncols> -F[f|s]<width(s)>/<height(s)>[+f<wfracs/hfracs>] [-A<autolabelinfo>]\n", name);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-C[<side>]<clearance>[u]] [%s] [-SC<layout>][+<mods>] [-SR<layout>][+<mods>]\n\t[-M<margins>] [%s] [-T<title>] [%s] [%s]\n\n", GMT_J_OPT, GMT_Rgeo_OPT, GMT_V_OPT, GMT_PAR_OPT);
+	GMT_Message (API, GMT_TIME_NONE, "usage: %s begin <nrows>x<ncols> -F[f|s]<width(s)>/<height(s)>[+f<wfracs/hfracs>][+c<gap>][+g<fill>][+p<pen>][+w<pen>] [-A<autolabelinfo>]\n", name);
+	GMT_Message (API, GMT_TIME_NONE, "\t[-C[<side>]<clearance>[u]] [%s] [-SC<layout>][+<mods>] [-SR<layout>][+<mods>]\n\t[-M<margins>] [%s] [-T<title>] [%s] [-W<pen>] [%s]\n\n", GMT_J_OPT, GMT_Rgeo_OPT, GMT_V_OPT, GMT_PAR_OPT);
```
Comment (seisman) [inline src/subplot.c]: **-W** need to be removed.

File: test/subplot/tictactoe.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+# Test lines between subplot with -W
+gmt begin tictactoe ps
+    gmt set FONT_ANNOT_PRIMARY 12p
+    gmt subplot begin 3x3 -Fs5c -M0 -W1p,- -Blrbt
```
Comment (seisman) [inline test/subplot/tictactoe.sh]:
```suggestion
    gmt subplot begin 3x3 -Fs5c+w1p,1 -M0 -Blrbt
```

**Comment by PaulWessel** [comment]
Hope I got them all?

File: doc/rst/source/subplot.rst_ [hunk 6]
```diff
@@ -105,10 +105,9 @@ Optional Arguments
 
 **-C**\ [*side*]\ /*clearance*\ [**u**\ ]
     Reserve a space of dimension *clearance* between the margin and the subplot on the specified
-    side, using *side* values from **w**, **e**, **s**, **n**, or **x** for both **w** and **e**
-    or **y** for both **s** and **n**.  The option is repeatable to set aside space
-    on more than one side.  If no *side* is given the it means the same constant clearance on all sides.
-    Such space will be left untouched by the main map plotting but can
+    side, using *side* values from **w**, **e**, **s**, or **n**, or **x** for both **w** and **e**
+    or **y** for both **s** and *n**.  No *side* means all sides. The option is repeatable to set aside space
```
Comment (seisman) [inline doc/rst/source/subplot.rst_]:
```suggestion
    or **y** for both **s** and **n**.  No *side* means all sides. The option is repeatable to set aside space
```

**Pull request merged**